scala-thoughts.md

```diff
@@ -0,0 +1,8 @@
+
+It is very easy to do recursion with Scala, I like how you have to declare the type when writing a recursive function. It is also very easy to create parameters to functions in Scala.
+
+I don't love the list creation in Scala. I was originally trying to get user input and put it into a list (I later switched to putting it into a string) and I was really struggling to understand what the rules regarding lists are. I would like to learn more about how lists work in Scala, and understand how one can edit a list. However, I did find it easy to work with strings, which was very nice.
+
+I would also like to learn about the type formatting in Scala. It seems like it was made with great intention. If I had a better understanding the type formatting I could use type information to my advantage when writing and debugging code.
```

Comment:
I agree that typing in Scala can be really helpful. As you said, recursive methods require return types, which makes it easy for reading the code and figuring out expected responses especially with longer functions. Since everything in Scala is supposed to have a type (as functions can be considered as values, and all values have types), we can write programs in functional style while having the option to annotate the code with types for easier understanding.